### PR TITLE
Multi-node+multi-GPU `papers100m`+`GCN` example

### PR DESCRIPTION
**Pull request by puririshi98**
working w/ nvidia pyg container

### CODE REVIEW

**Review by rusty1s — COMMENTED** [review]
Wondering if it is possible to merge this with the initial multi-GPU example, or whether there exists a strong reason not do this.

**Comment by puririshi98** [comment]
> Wondering if it is possible to merge this with the initial multi-GPU example, or whether there exists a strong reason not do this.

the set up is relatively different(the old one uses mp spawn for single node multigpu, for multinode multigpu we are using torch distributed w/ nccl backend, i think for now we should keep them seperate

**Review by akihironitta — COMMENTED** [review]
This will be a very nice addition! 🚀

Haven't tried running this example in a multi-node multi-gpu environment myself, but the code looks good to me. Just like Matthias minimised the other example in #7954, I think we should keep this example minimal, too, where it makes sense.

File: examples/multi_gpu/multigpu_papers100m_gcn.py [hunk 1]
```diff
@@ -14,15 +14,15 @@
 from torch_geometric.nn import GCNConv
 
 
-def pyg_num_work():
+def pyg_num_work(world_size):
     num_work = None
     if hasattr(os, "sched_getaffinity"):
         try:
-            num_work = len(os.sched_getaffinity(0)) / 2
+            num_work = len(os.sched_getaffinity(0)) / (2 * world_size)
```
[the remark this answers is not in the document]
Comment (akihironitta) [reply examples/multi_gpu/multigpu_papers100m_gcn.py:21]: nvm this is a script for single-node multi-gpu environments :)

File: examples/multi_gpu/multigpu_papers100m_gcn.py [hunk 2]
```diff
@@ -14,15 +14,15 @@
 from torch_geometric.nn import GCNConv
```
Comment (akihironitta) [inline examples/multi_gpu/multigpu_papers100m_gcn.py:14]: (nit: Feel free to ignore. I'd name this file `singlenode_multigpu_papers100m_gcn.py`)

Comment (puririshi98) [reply examples/multi_gpu/multigpu_papers100m_gcn.py:14]: agreed

**Comment by puririshi98** [comment]
@akihironitta @rusty1s lmk if anything else needed to merge

**Comment by puririshi98** [comment]
@akihironitta @rusty1s lmk if anything else needed to merge

**Review by rusty1s — APPROVED** [review]
Thanks @puririshi98. I cleaned it up a bit. I really like the example. One thing we should definitely improve on would be to allow distributed evaluation as well, but it is not blocking this PR.

**Pull request merged**